systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -308,13 +317,164 @@ void testKafkaNodePoolBrokerIdsManagementUsingAnnotations(ExtensionContext exten
+                    .addToLimits("memory", new Quantity(knpBAndCLimitMemoryQuantityConfig)) // overriding resource.limit
+                .build())
+            .endSpec()
+            .build();
+
+        LOGGER.info("Verify correct config. propagation from Kafka to Kafka Node Pool Pods");
+
+        resourceManager.createResourceWithWait(extensionContext, poolA, poolB, poolC, kafka);
+        waitForPodsReady(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), 4, true);
+
+        // assert jvm and limits config (written in kafka) is propagated into KNP A Pods
+        verifyResourcesConfigurationPropagatedToKnpPods(testStorage, nodePoolNameA, kafkaJvmXmxConfig, kafkaLimitMemoryQuantityConfig);
+
+        // assert jvm and limits config (written in kafka) is overwritten by respective config present in KNP B Pods
+        verifyResourcesConfigurationPropagatedToKnpPods(testStorage, nodePoolNameB, knpBJvmXmxConfig, knpBAndCLimitMemoryQuantityConfig);
+
+        // assert jvm config (written in kafka) is propagated into KNP C Pods while limit configuration not, as it is present in KNP C.
+        verifyResourcesConfigurationPropagatedToKnpPods(testStorage, nodePoolNameC, kafkaJvmXmxConfig, knpBAndCLimitMemoryQuantityConfig);
+
+        LOGGER.info("Verify correct config. propagation when modifying kafka CR");
+
+        final Map<String, String> nodePoolAPodsSnapshot = PodUtils.podSnapshot(testStorage.getNamespaceName(), nodePoolALabelSelector);
+        final Map<String, String> nodePoolBPodsSnapshot = PodUtils.podSnapshot(testStorage.getNamespaceName(), nodePoolBLabelSelector);
+
+        // modify resource.limit in kafka CR
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), k -> {
+            k.getSpec().getKafka().getResources().getLimits().replace("memory", new Quantity(kafkaLimitMemoryQuantityConfigModified));
+        }, testStorage.getNamespaceName());
+
+        // (as KNP A is the only one that inherit it from Kafka it is the only one which needs to be updated by this change
+        RollingUpdateUtils.waitForNoRollingUpdate(testStorage.getNamespaceName(), nodePoolBLabelSelector, nodePoolBPodsSnapshot);
+        RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), nodePoolALabelSelector, 2, nodePoolAPodsSnapshot);
+        //waitForPodsReady(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), scaledUpBrokerReplicaCount, false);
+
+        // assert that new changed configuration is present in pods under Kafka Node Pool A
+        verifyResourcesConfigurationPropagatedToKnpPods(testStorage, nodePoolNameA, kafkaJvmXmxConfig, kafkaLimitMemoryQuantityConfigModified);
+
+        LOGGER.info("Modifying configuration relevant or irrelevant based on role of given KafkaNodePools");
+
+        final Pod podFromKnpC = PodUtils.getPodsByPrefixInNameWithDynamicWait(testStorage.getNamespaceName(), testStorage.getClusterName() + "-" + nodePoolNameC).get(0);
```

Comment:
Isn't there some method for constructing the StrimziPodSet name instead of doing `testStorage.getClusterName() + "-" + nodePoolNameC`?